### DISCUSSION TITLE
So far the best HomeKit solution 👌🏻

### DISCUSSION THREAD

**Discussion by rakeshnsw** [HomeSpan/HomeSpan#598] [Show and tell]
I built this prototype just to try HomeSpan to control my two water sprinklers. Honestly, this is now the most stable and reliable switch across all other fancy and expensive home automation devices I have. I am planning to flash HomeSpan on all other compatible devices. Thanks a lot for developing this awesome solution. 🙏🏻![image](https://github.com/HomeSpan/HomeSpan/assets/52046964/4405d6ff-ca0b-4943-ba1b-f751fb2cf0d2)

**Comment by HomeSpan** [comment]
Thanks so much for the comment!  I'm really glad you are finding the library useful.

**Reply by kowalenzo** [reply]
Thank you for developing this library! My garden irrigation system also runs on your code, 3 relays and retention tank level monitor disguised as humidity sensor :) Also garage door opener, electromagnetic locks, etc. Wonderful stuff. 
I do experience wifi connectivity issues (possibly due to running a mesh wifi network), but it's not a deal breaker. In the long term I plan on moving all systems to thread network based on esp32-h2 or c6. Hope it will solve the wifi reliability issue.

**Reply by HomeSpan** [reply]
I'm looking forward to Espressif releasing an Arduino library for the H2.  Will be very interested in what can be done with thread networks.
